python/triton/language/core.py

```diff
@@ -514,6 +535,17 @@ def to_ir(self, builder: ir.builder):
 # -----------------------
 
 
+class const:
+    """
+    This class is used as a type annotation to mark pointers to tensors
+    that cannot be modified. Store cannot be called with a const pointer.
+    Const pointers are represented by `const_pointer_type`, and usual
+    Triton type consistency rules apply - you cannot return a const pointer
+    from a function that returns also a non-const pointer, etc.
+    """
```

Comment:
I don't think this description is very clear. Triton does not see tensors -- only pointers. Could we express this in terms of aliasing?

Reply:
Would rewriting first sentence like this help?
"This class is used as a type annotation to mark pointers to constant data."
And instead of "const pointer" use "pointer to const"?